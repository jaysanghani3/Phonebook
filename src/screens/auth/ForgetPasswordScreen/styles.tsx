import {StyleSheet} from 'react-native';
import Colors from '../../../utils/Colors';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.white,
    padding: 25,
  },
  welcomeHeader: {
    fontSize: 30,
    fontWeight: 'bold',
    color: Colors.black,
    textAlign: 'center',
  },
  formBox: {
    marginVertical: 50,
  },
  textArea: {
    height: 50,
    paddingStart: 10,
    borderRadius: 8,
    borderColor: Colors.gray,
    borderWidth: 1,
    marginVertical: 5,
  },
  redText: {
    color: Colors.error,
  },
  icon: {
    width: 350,
    height: 350,
  },
});

export default styles;
