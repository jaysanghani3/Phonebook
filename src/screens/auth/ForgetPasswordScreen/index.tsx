import {firebase} from '@react-native-firebase/firestore';
import React, {useState} from 'react';
import {
  Text,
  View,
  TouchableOpacity,
  TextInput,
  Image,
  TouchableWithoutFeedback,
  Keyboard,
  ScrollView,
} from 'react-native';
import styles from './styles';
import Toast from 'react-native-simple-toast';
import globalStyles from '../../../utils/globalStyles';
import Loader from '../../../components/Loader';
import Icons from '../../../assets';
import {isEmailValid} from '../../../utils/validations';
import Strings from '../../../translations/Strings';

import {NativeStackScreenProps} from '@react-navigation/native-stack';
import {RouteStackParamsList} from '../../../navigators/AppNavigators';
import {Routes} from '../../../navigators/routes';

type ForgetPasswordScreenProps = NativeStackScreenProps<
  RouteStackParamsList,
  Routes.ForgetPasswordScreen
>;

const ForgetPasswordScreen = ({navigation}: ForgetPasswordScreenProps) => {
  const [email, setEmail] = useState('');
  const [isLoading, setIsLoading] = useState(false);
  const [error, setError] = useState('');

  const forgetPassword = async () => {
    try {
      if (!email) {
        setError(Strings.required);
        return;
      }
      if (!isEmailValid(email)) {
        setError(Strings.invalidEmail);
        return;
      }
      setError('');
      setIsLoading(true);
      await firebase.auth().sendPasswordResetEmail(email);
      setIsLoading(false);

      Toast.show(Strings.linkSend, Toast.LONG);
      navigation.navigate('LoginScreen');
    } catch (e: any) {
      const err = e.code.split('/')[1].split('-').join(' ');
      Toast.show(err, Toast.LONG);
      setEmail('');
      setIsLoading(false);
    }
  };
  return (
    <TouchableWithoutFeedback onPress={() => Keyboard.dismiss()}>
      <ScrollView style={styles.container}>
        <Loader state={isLoading} />
        <View style={globalStyles.alignCenter}>
          <Image source={Icons.forget} style={styles.icon} />
        </View>

        <Text style={styles.welcomeHeader}>{Strings.forgotText}</Text>

        <View style={styles.formBox}>
          <TextInput
            inputMode="email"
            style={styles.textArea}
            placeholder={Strings.emailPlaceholder}
            defaultValue={email}
            onChangeText={newEmail => setEmail(newEmail)}
          />
          <Text style={styles.redText}>{error}</Text>
        </View>

        <TouchableOpacity style={globalStyles.btn} onPress={forgetPassword}>
          <Text style={globalStyles.btnText}>{Strings.passwordResetBtn}</Text>
        </TouchableOpacity>
      </ScrollView>
    </TouchableWithoutFeedback>
  );
};

export default ForgetPasswordScreen;
