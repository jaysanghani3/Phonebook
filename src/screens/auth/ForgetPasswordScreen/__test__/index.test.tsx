import renderer from 'react-test-renderer';
import ForgetPasswordScreen from '../index';

it('renders correctly', () => {
  const tree = renderer
    .create(<ForgetPasswordScreen navigation={undefined} route={undefined} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
