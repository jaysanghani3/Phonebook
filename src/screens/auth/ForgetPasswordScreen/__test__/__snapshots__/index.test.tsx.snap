// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "padding": 25,
    }
  }
>
  <View>
    <View
      style={
        {
          "alignItems": "center",
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../src/assets/forget.jpg",
          }
        }
        style={
          {
            "height": 350,
            "width": 350,
          }
        }
      />
    </View>
    <Text
      style={
        {
          "color": "#000000",
          "fontSize": 30,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Forgot Password
    </Text>
    <View
      style={
        {
          "marginVertical": 50,
        }
      }
    >
      <TextInput
        defaultValue=""
        inputMode="email"
        onChangeText={[Function]}
        placeholder="Email"
        style={
          {
            "borderColor": "#8D8D8D",
            "borderRadius": 8,
            "borderWidth": 1,
            "height": 50,
            "marginVertical": 5,
            "paddingStart": 10,
          }
        }
      />
      <Text
        style={
          {
            "color": "#FF0000",
          }
        }
      />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#66347f",
          "borderRadius": 8,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Password Reset
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
